include/nn/bcat.h

```diff
@@ -0,0 +1,109 @@
+/**
+ * @file bcat.h
+ * @brief BCAT service implementation.
+ */
+
+#pragma once
+
+#include "types.h"
+#include "os.h"
+
+namespace nn {
+
+    namespace bcat {
+
+        struct DirectoryName {
+            void isValid();
+        };
+
+        struct FileName {
+            void isValid(); 
+        };
+
+        class DeliveryCacheDirectory {
+            DeliveryCacheDirectory();
+            ~DeliveryCacheDirectory();
+            //void Open(char const*);
+            Result Open(nn::bcat::DirectoryName const&);
+            void GetCount();
+            void Close();
+        }; // namespace DeliveryCacheDirectory
+
+        class DeliveryCacheFile {
+            DeliveryCacheFile();
+            ~DeliveryCacheFile();
+            Result Open(char const*);
+            void Read(u64* file1, s64, void *, u64 file2); // probably better names
```

Comment:
Are you sure they used u64 here? I think it's more likely to be size_t. (Keep in mind that typedefs don't show up in symbols)

The comment should probably also be removed.